### ISSUE TITLE
StoredValue & DirectValue tests fail in debug mode

### ISSUE THREAD

**Issue by j-o** [cginternals/cppexpose#11]
```
[==========] Running 128 tests from 42 test cases.
[----------] Global test environment set-up.
[----------] 10 tests from variantTest
[ RUN      ] variantTest.constructorBool
[       OK ] variantTest.constructorBool (0 ms)
[ RUN      ] variantTest.constructorString
[       OK ] variantTest.constructorString (1 ms)
[ RUN      ] variantTest.constructorSignedIntegral
[       OK ] variantTest.constructorSignedIntegral (0 ms)
[ RUN      ] variantTest.constructorUnsignedIntegral
[       OK ] variantTest.constructorUnsignedIntegral (0 ms)
[ RUN      ] variantTest.constructorFloat
[       OK ] variantTest.constructorFloat (0 ms)
[ RUN      ] variantTest.constructorEmpty
[       OK ] variantTest.constructorEmpty (0 ms)
[ RUN      ] variantTest.constructorVariantArray
[       OK ] variantTest.constructorVariantArray (1 ms)
[ RUN      ] variantTest.staticConstructorVariantArray
[       OK ] variantTest.staticConstructorVariantArray (0 ms)
[ RUN      ] variantTest.constructorVariantMap
[       OK ] variantTest.constructorVariantMap (0 ms)
[ RUN      ] variantTest.staticConstructorVariantMap
[       OK ] variantTest.staticConstructorVariantMap (0 ms)
[----------] 10 tests from variantTest (2 ms total)

[----------] 5 tests from PropertyInstantiation_test/0, where TypeParam = bool
[ RUN      ] PropertyInstantiation_test/0.instantiatePropertyWith_String_LambdaGetter_LambdaSetter
[       OK ] PropertyInstantiation_test/0.instantiatePropertyWith_String_LambdaGetter_LambdaSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/0.instantiatePropertyWith_String_StaticGetter_StaticSetter
[       OK ] PropertyInstantiation_test/0.instantiatePropertyWith_String_StaticGetter_StaticSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/0.instantiatePropertyWith_String_Object_GetterConst_SetterConst
[       OK ] PropertyInstantiation_test/0.instantiatePropertyWith_String_Object_GetterConst_SetterConst (1 ms)
[ RUN      ] PropertyInstantiation_test/0.instantiateConstPropertyWith_String_LambdaGetter
[       OK ] PropertyInstantiation_test/0.instantiateConstPropertyWith_String_LambdaGetter (0 ms)
[ RUN      ] PropertyInstantiation_test/0.instantiateConstPropertyWith_String_StaticGetter
[       OK ] PropertyInstantiation_test/0.instantiateConstPropertyWith_String_StaticGetter (1 ms)
[----------] 5 tests from PropertyInstantiation_test/0 (3 ms total)

[----------] 5 tests from PropertyInstantiation_test/1, where TypeParam = float
[ RUN      ] PropertyInstantiation_test/1.instantiatePropertyWith_String_LambdaGetter_LambdaSetter
[       OK ] PropertyInstantiation_test/1.instantiatePropertyWith_String_LambdaGetter_LambdaSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/1.instantiatePropertyWith_String_StaticGetter_StaticSetter
[       OK ] PropertyInstantiation_test/1.instantiatePropertyWith_String_StaticGetter_StaticSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/1.instantiatePropertyWith_String_Object_GetterConst_SetterConst
[       OK ] PropertyInstantiation_test/1.instantiatePropertyWith_String_Object_GetterConst_SetterConst (1 ms)
[ RUN      ] PropertyInstantiation_test/1.instantiateConstPropertyWith_String_LambdaGetter
[       OK ] PropertyInstantiation_test/1.instantiateConstPropertyWith_String_LambdaGetter (0 ms)
[ RUN      ] PropertyInstantiation_test/1.instantiateConstPropertyWith_String_StaticGetter
[       OK ] PropertyInstantiation_test/1.instantiateConstPropertyWith_String_StaticGetter (1 ms)
[----------] 5 tests from PropertyInstantiation_test/1 (3 ms total)

[----------] 5 tests from PropertyInstantiation_test/2, where TypeParam = double
[ RUN      ] PropertyInstantiation_test/2.instantiatePropertyWith_String_LambdaGetter_LambdaSetter
[       OK ] PropertyInstantiation_test/2.instantiatePropertyWith_String_LambdaGetter_LambdaSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/2.instantiatePropertyWith_String_StaticGetter_StaticSetter
[       OK ] PropertyInstantiation_test/2.instantiatePropertyWith_String_StaticGetter_StaticSetter (1 ms)
[ RUN      ] PropertyInstantiation_test/2.instantiatePropertyWith_String_Object_GetterConst_SetterConst
[       OK ] PropertyInstantiation_test/2.instantiatePropertyWith_String_Object_GetterConst_SetterConst (1 ms)
[ RUN      ] PropertyInstantiation_test/2.instantiateConstPropertyWith_String_LambdaGetter
[       OK ] PropertyInstantiation_test/2.instantiateConstPropertyWith_String_LambdaGetter (0 ms)
[ RUN      ] PropertyInstantiation_test/2.instantiateConstPropertyWith_String_StaticGetter
[       OK ] PropertyInstantiation_test/2.instantiateConstPropertyWith_String_StaticGetter (1 ms)
[----------] 5 tests from PropertyInstantiation_test/2 (3 ms total)

[----------] 5 tests from PropertyInstantiation_test/3, where TypeParam = signed char
[ RUN      ] PropertyInstantiation_test/3.instantiatePropertyWith_String_LambdaGetter_LambdaSetter
[       OK ] PropertyInstantiation_test/3.instantiatePropertyWith_String_LambdaGetter_LambdaSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/3.instantiatePropertyWith_String_StaticGetter_StaticSetter
[       OK ] PropertyInstantiation_test/3.instantiatePropertyWith_String_StaticGetter_StaticSetter (1 ms)
[ RUN      ] PropertyInstantiation_test/3.instantiatePropertyWith_String_Object_GetterConst_SetterConst
[       OK ] PropertyInstantiation_test/3.instantiatePropertyWith_String_Object_GetterConst_SetterConst (0 ms)
[ RUN      ] PropertyInstantiation_test/3.instantiateConstPropertyWith_String_LambdaGetter
[       OK ] PropertyInstantiation_test/3.instantiateConstPropertyWith_String_LambdaGetter (0 ms)
[ RUN      ] PropertyInstantiation_test/3.instantiateConstPropertyWith_String_StaticGetter
[       OK ] PropertyInstantiation_test/3.instantiateConstPropertyWith_String_StaticGetter (0 ms)
[----------] 5 tests from PropertyInstantiation_test/3 (3 ms total)

[----------] 5 tests from PropertyInstantiation_test/4, where TypeParam = unsigned char
[ RUN      ] PropertyInstantiation_test/4.instantiatePropertyWith_String_LambdaGetter_LambdaSetter
[       OK ] PropertyInstantiation_test/4.instantiatePropertyWith_String_LambdaGetter_LambdaSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/4.instantiatePropertyWith_String_StaticGetter_StaticSetter
[       OK ] PropertyInstantiation_test/4.instantiatePropertyWith_String_StaticGetter_StaticSetter (1 ms)
[ RUN      ] PropertyInstantiation_test/4.instantiatePropertyWith_String_Object_GetterConst_SetterConst
[       OK ] PropertyInstantiation_test/4.instantiatePropertyWith_String_Object_GetterConst_SetterConst (0 ms)
[ RUN      ] PropertyInstantiation_test/4.instantiateConstPropertyWith_String_LambdaGetter
[       OK ] PropertyInstantiation_test/4.instantiateConstPropertyWith_String_LambdaGetter (1 ms)
[ RUN      ] PropertyInstantiation_test/4.instantiateConstPropertyWith_String_StaticGetter
[       OK ] PropertyInstantiation_test/4.instantiateConstPropertyWith_String_StaticGetter (0 ms)
[----------] 5 tests from PropertyInstantiation_test/4 (2 ms total)

[----------] 5 tests from PropertyInstantiation_test/5, where TypeParam = short
[ RUN      ] PropertyInstantiation_test/5.instantiatePropertyWith_String_LambdaGetter_LambdaSetter
[       OK ] PropertyInstantiation_test/5.instantiatePropertyWith_String_LambdaGetter_LambdaSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/5.instantiatePropertyWith_String_StaticGetter_StaticSetter
[       OK ] PropertyInstantiation_test/5.instantiatePropertyWith_String_StaticGetter_StaticSetter (1 ms)
[ RUN      ] PropertyInstantiation_test/5.instantiatePropertyWith_String_Object_GetterConst_SetterConst
[       OK ] PropertyInstantiation_test/5.instantiatePropertyWith_String_Object_GetterConst_SetterConst (0 ms)
[ RUN      ] PropertyInstantiation_test/5.instantiateConstPropertyWith_String_LambdaGetter
[       OK ] PropertyInstantiation_test/5.instantiateConstPropertyWith_String_LambdaGetter (1 ms)
[ RUN      ] PropertyInstantiation_test/5.instantiateConstPropertyWith_String_StaticGetter
[       OK ] PropertyInstantiation_test/5.instantiateConstPropertyWith_String_StaticGetter (0 ms)
[----------] 5 tests from PropertyInstantiation_test/5 (2 ms total)

[----------] 5 tests from PropertyInstantiation_test/6, where TypeParam = unsigned short
[ RUN      ] PropertyInstantiation_test/6.instantiatePropertyWith_String_LambdaGetter_LambdaSetter
[       OK ] PropertyInstantiation_test/6.instantiatePropertyWith_String_LambdaGetter_LambdaSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/6.instantiatePropertyWith_String_StaticGetter_StaticSetter
[       OK ] PropertyInstantiation_test/6.instantiatePropertyWith_String_StaticGetter_StaticSetter (1 ms)
[ RUN      ] PropertyInstantiation_test/6.instantiatePropertyWith_String_Object_GetterConst_SetterConst
[       OK ] PropertyInstantiation_test/6.instantiatePropertyWith_String_Object_GetterConst_SetterConst (0 ms)
[ RUN      ] PropertyInstantiation_test/6.instantiateConstPropertyWith_String_LambdaGetter
[       OK ] PropertyInstantiation_test/6.instantiateConstPropertyWith_String_LambdaGetter (1 ms)
[ RUN      ] PropertyInstantiation_test/6.instantiateConstPropertyWith_String_StaticGetter
[       OK ] PropertyInstantiation_test/6.instantiateConstPropertyWith_String_StaticGetter (0 ms)
[----------] 5 tests from PropertyInstantiation_test/6 (2 ms total)

[----------] 5 tests from PropertyInstantiation_test/7, where TypeParam = int
[ RUN      ] PropertyInstantiation_test/7.instantiatePropertyWith_String_LambdaGetter_LambdaSetter
[       OK ] PropertyInstantiation_test/7.instantiatePropertyWith_String_LambdaGetter_LambdaSetter (1 ms)
[ RUN      ] PropertyInstantiation_test/7.instantiatePropertyWith_String_StaticGetter_StaticSetter
[       OK ] PropertyInstantiation_test/7.instantiatePropertyWith_String_StaticGetter_StaticSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/7.instantiatePropertyWith_String_Object_GetterConst_SetterConst
[       OK ] PropertyInstantiation_test/7.instantiatePropertyWith_String_Object_GetterConst_SetterConst (1 ms)
[ RUN      ] PropertyInstantiation_test/7.instantiateConstPropertyWith_String_LambdaGetter
[       OK ] PropertyInstantiation_test/7.instantiateConstPropertyWith_String_LambdaGetter (1 ms)
[ RUN      ] PropertyInstantiation_test/7.instantiateConstPropertyWith_String_StaticGetter
[       OK ] PropertyInstantiation_test/7.instantiateConstPropertyWith_String_StaticGetter (0 ms)
[----------] 5 tests from PropertyInstantiation_test/7 (3 ms total)

[----------] 5 tests from PropertyInstantiation_test/8, where TypeParam = unsigned int
[ RUN      ] PropertyInstantiation_test/8.instantiatePropertyWith_String_LambdaGetter_LambdaSetter
[       OK ] PropertyInstantiation_test/8.instantiatePropertyWith_String_LambdaGetter_LambdaSetter (1 ms)
[ RUN      ] PropertyInstantiation_test/8.instantiatePropertyWith_String_StaticGetter_StaticSetter
[       OK ] PropertyInstantiation_test/8.instantiatePropertyWith_String_StaticGetter_StaticSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/8.instantiatePropertyWith_String_Object_GetterConst_SetterConst
[       OK ] PropertyInstantiation_test/8.instantiatePropertyWith_String_Object_GetterConst_SetterConst (1 ms)
[ RUN      ] PropertyInstantiation_test/8.instantiateConstPropertyWith_String_LambdaGetter
[       OK ] PropertyInstantiation_test/8.instantiateConstPropertyWith_String_LambdaGetter (0 ms)
[ RUN      ] PropertyInstantiation_test/8.instantiateConstPropertyWith_String_StaticGetter
[       OK ] PropertyInstantiation_test/8.instantiateConstPropertyWith_String_StaticGetter (1 ms)
[----------] 5 tests from PropertyInstantiation_test/8 (3 ms total)

[----------] 5 tests from PropertyInstantiation_test/9, where TypeParam = __int64
[ RUN      ] PropertyInstantiation_test/9.instantiatePropertyWith_String_LambdaGetter_LambdaSetter
[       OK ] PropertyInstantiation_test/9.instantiatePropertyWith_String_LambdaGetter_LambdaSetter (1 ms)
[ RUN      ] PropertyInstantiation_test/9.instantiatePropertyWith_String_StaticGetter_StaticSetter
[       OK ] PropertyInstantiation_test/9.instantiatePropertyWith_String_StaticGetter_StaticSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/9.instantiatePropertyWith_String_Object_GetterConst_SetterConst
[       OK ] PropertyInstantiation_test/9.instantiatePropertyWith_String_Object_GetterConst_SetterConst (1 ms)
[ RUN      ] PropertyInstantiation_test/9.instantiateConstPropertyWith_String_LambdaGetter
[       OK ] PropertyInstantiation_test/9.instantiateConstPropertyWith_String_LambdaGetter (0 ms)
[ RUN      ] PropertyInstantiation_test/9.instantiateConstPropertyWith_String_StaticGetter
[       OK ] PropertyInstantiation_test/9.instantiateConstPropertyWith_String_StaticGetter (1 ms)
[----------] 5 tests from PropertyInstantiation_test/9 (3 ms total)

[----------] 5 tests from PropertyInstantiation_test/10, where TypeParam = unsigned __int64
[ RUN      ] PropertyInstantiation_test/10.instantiatePropertyWith_String_LambdaGetter_LambdaSetter
[       OK ] PropertyInstantiation_test/10.instantiatePropertyWith_String_LambdaGetter_LambdaSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/10.instantiatePropertyWith_String_StaticGetter_StaticSetter
[       OK ] PropertyInstantiation_test/10.instantiatePropertyWith_String_StaticGetter_StaticSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/10.instantiatePropertyWith_String_Object_GetterConst_SetterConst
[       OK ] PropertyInstantiation_test/10.instantiatePropertyWith_String_Object_GetterConst_SetterConst (1 ms)
[ RUN      ] PropertyInstantiation_test/10.instantiateConstPropertyWith_String_LambdaGetter
[       OK ] PropertyInstantiation_test/10.instantiateConstPropertyWith_String_LambdaGetter (0 ms)
[ RUN      ] PropertyInstantiation_test/10.instantiateConstPropertyWith_String_StaticGetter
[       OK ] PropertyInstantiation_test/10.instantiateConstPropertyWith_String_StaticGetter (1 ms)
[----------] 5 tests from PropertyInstantiation_test/10 (3 ms total)

[----------] 5 tests from PropertyInstantiation_test/11, where TypeParam = class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >
[ RUN      ] PropertyInstantiation_test/11.instantiatePropertyWith_String_LambdaGetter_LambdaSetter
[       OK ] PropertyInstantiation_test/11.instantiatePropertyWith_String_LambdaGetter_LambdaSetter (0 ms)
[ RUN      ] PropertyInstantiation_test/11.instantiatePropertyWith_String_StaticGetter_StaticSetter
[       OK ] PropertyInstantiation_test/11.instantiatePropertyWith_String_StaticGetter_StaticSetter (1 ms)
[ RUN      ] PropertyInstantiation_test/11.instantiatePropertyWith_String_Object_GetterConst_SetterConst
[       OK ] PropertyInstantiation_test/11.instantiatePropertyWith_String_Object_GetterConst_SetterConst (0 ms)
[ RUN      ] PropertyInstantiation_test/11.instantiateConstPropertyWith_String_LambdaGetter
[       OK ] PropertyInstantiation_test/11.instantiateConstPropertyWith_String_LambdaGetter (0 ms)
[ RUN      ] PropertyInstantiation_test/11.instantiateConstPropertyWith_String_StaticGetter
[       OK ] PropertyInstantiation_test/11.instantiateConstPropertyWith_String_StaticGetter (0 ms)
[----------] 5 tests from PropertyInstantiation_test/11 (3 ms total)

[----------] 12 tests from PropertyTest
[ RUN      ] PropertyTest.boolSet
[       OK ] PropertyTest.boolSet (0 ms)
[ RUN      ] PropertyTest.ArraySet
[       OK ] PropertyTest.ArraySet (1 ms)
[ RUN      ] PropertyTest.stringSet
[       OK ] PropertyTest.stringSet (0 ms)
[ RUN      ] PropertyTest.conversionTest_int
[       OK ] PropertyTest.conversionTest_int (1 ms)
[ RUN      ] PropertyTest.conversionTest_negative_int
[       OK ] PropertyTest.conversionTest_negative_int (1 ms)
[ RUN      ] PropertyTest.conversionTest_string
[       OK ] PropertyTest.conversionTest_string (0 ms)
[ RUN      ] PropertyTest.conversionTest_string_variant
[       OK ] PropertyTest.conversionTest_string_variant (1 ms)
[ RUN      ] PropertyTest.typesBool
[       OK ] PropertyTest.typesBool (0 ms)
[ RUN      ] PropertyTest.typesSignedIntegral
[       OK ] PropertyTest.typesSignedIntegral (1 ms)
[ RUN      ] PropertyTest.typesUnsignedIntegral
[       OK ] PropertyTest.typesUnsignedIntegral (0 ms)
[ RUN      ] PropertyTest.typesString
[       OK ] PropertyTest.typesString (1 ms)
[ RUN      ] PropertyTest.typesFloat
[       OK ] PropertyTest.typesFloat (0 ms)
[----------] 12 tests from PropertyTest (7 ms total)

[----------] 1 test from DirectValueInstantiation_test/0, where TypeParam = bool
[ RUN      ] DirectValueInstantiation_test/0.instantiateDirectValue
[       OK ] DirectValueInstantiation_test/0.instantiateDirectValue (0 ms)
[----------] 1 test from DirectValueInstantiation_test/0 (0 ms total)

[----------] 1 test from DirectValueInstantiation_test/1, where TypeParam = float
[ RUN      ] DirectValueInstantiation_test/1.instantiateDirectValue
[       OK ] DirectValueInstantiation_test/1.instantiateDirectValue (0 ms)
[----------] 1 test from DirectValueInstantiation_test/1 (0 ms total)

[----------] 1 test from DirectValueInstantiation_test/2, where TypeParam = double
[ RUN      ] DirectValueInstantiation_test/2.instantiateDirectValue
[       OK ] DirectValueInstantiation_test/2.instantiateDirectValue (0 ms)
[----------] 1 test from DirectValueInstantiation_test/2 (0 ms total)

[----------] 1 test from DirectValueInstantiation_test/3, where TypeParam = signed char
[ RUN      ] DirectValueInstantiation_test/3.instantiateDirectValue
[       OK ] DirectValueInstantiation_test/3.instantiateDirectValue (0 ms)
[----------] 1 test from DirectValueInstantiation_test/3 (0 ms total)

[----------] 1 test from DirectValueInstantiation_test/4, where TypeParam = unsigned char
[ RUN      ] DirectValueInstantiation_test/4.instantiateDirectValue
[       OK ] DirectValueInstantiation_test/4.instantiateDirectValue (0 ms)
[----------] 1 test from DirectValueInstantiation_test/4 (0 ms total)

[----------] 1 test from DirectValueInstantiation_test/5, where TypeParam = short
[ RUN      ] DirectValueInstantiation_test/5.instantiateDirectValue
[       OK ] DirectValueInstantiation_test/5.instantiateDirectValue (0 ms)
[----------] 1 test from DirectValueInstantiation_test/5 (0 ms total)

[----------] 1 test from DirectValueInstantiation_test/6, where TypeParam = unsigned short
[ RUN      ] DirectValueInstantiation_test/6.instantiateDirectValue
[       OK ] DirectValueInstantiation_test/6.instantiateDirectValue (0 ms)
[----------] 1 test from DirectValueInstantiation_test/6 (0 ms total)

[----------] 1 test from DirectValueInstantiation_test/7, where TypeParam = int
[ RUN      ] DirectValueInstantiation_test/7.instantiateDirectValue
[       OK ] DirectValueInstantiation_test/7.instantiateDirectValue (0 ms)
[----------] 1 test from DirectValueInstantiation_test/7 (0 ms total)

[----------] 1 test from DirectValueInstantiation_test/8, where TypeParam = unsigned int
[ RUN      ] DirectValueInstantiation_test/8.instantiateDirectValue
[       OK ] DirectValueInstantiation_test/8.instantiateDirectValue (0 ms)
[----------] 1 test from DirectValueInstantiation_test/8 (1 ms total)

[----------] 1 test from DirectValueInstantiation_test/9, where TypeParam = __int64
[ RUN      ] DirectValueInstantiation_test/9.instantiateDirectValue
[       OK ] DirectValueInstantiation_test/9.instantiateDirectValue (0 ms)
[----------] 1 test from DirectValueInstantiation_test/9 (0 ms total)

[----------] 1 test from DirectValueInstantiation_test/10, where TypeParam = unsigned __int64
[ RUN      ] DirectValueInstantiation_test/10.instantiateDirectValue
[       OK ] DirectValueInstantiation_test/10.instantiateDirectValue (0 ms)
[----------] 1 test from DirectValueInstantiation_test/10 (0 ms total)

[----------] 1 test from DirectValueInstantiation_test/11, where TypeParam = class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >
[ RUN      ] DirectValueInstantiation_test/11.instantiateDirectValue
[       OK ] DirectValueInstantiation_test/11.instantiateDirectValue (0 ms)
[----------] 1 test from DirectValueInstantiation_test/11 (0 ms total)

[----------] 1 test from DirectValueInstantiation_test/12, where TypeParam = class std::array<int,4>
[ RUN      ] DirectValueInstantiation_test/12.instantiateDirectValue
[       OK ] DirectValueInstantiation_test/12.instantiateDirectValue (0 ms)
[----------] 1 test from DirectValueInstantiation_test/12 (0 ms total)

[----------] 10 tests from DirectValueTest
[ RUN      ] DirectValueTest.boolSet
[       OK ] DirectValueTest.boolSet (1 ms)
[ RUN      ] DirectValueTest.stringSet
[       OK ] DirectValueTest.stringSet (0 ms)
[ RUN      ] DirectValueTest.arrayGet
[       OK ] DirectValueTest.arrayGet (0 ms)
[ RUN      ] DirectValueTest.arraySet
[       OK ] DirectValueTest.arraySet (0 ms)
[ RUN      ] DirectValueTest.constSet
[       OK ] DirectValueTest.constSet (0 ms)
[ RUN      ] DirectValueTest.typesBool
D:\projects\cppexpose\source\tests\cppexpose-test\DirectValueTest.cpp(59): error: Value of: test
  Actual: true
Expected: false
[  FAILED  ] DirectValueTest.typesBool (1 ms)
[ RUN      ] DirectValueTest.typesSignedIntegral
D:\projects\cppexpose\source\tests\cppexpose-test\DirectValueTest.cpp(59): error: Value of: test
  Actual: true
Expected: false
[  FAILED  ] DirectValueTest.typesSignedIntegral (0 ms)
[ RUN      ] DirectValueTest.typesUnsignedIntegral
D:\projects\cppexpose\source\tests\cppexpose-test\DirectValueTest.cpp(59): error: Value of: test
  Actual: true
Expected: false
[  FAILED  ] DirectValueTest.typesUnsignedIntegral (0 ms)
[ RUN      ] DirectValueTest.typesString
D:\projects\cppexpose\source\tests\cppexpose-test\DirectValueTest.cpp(59): error: Value of: test
  Actual: true
Expected: false
[  FAILED  ] DirectValueTest.typesString (1 ms)
[ RUN      ] DirectValueTest.typesFloat
D:\projects\cppexpose\source\tests\cppexpose-test\DirectValueTest.cpp(59): error: Value of: test
  Actual: true
Expected: false
[  FAILED  ] DirectValueTest.typesFloat (0 ms)
[----------] 10 tests from DirectValueTest (3 ms total)

[----------] 1 test from StoredValueInstantiation_test/0, where TypeParam = bool
[ RUN      ] StoredValueInstantiation_test/0.instantiateStoredValue
[       OK ] StoredValueInstantiation_test/0.instantiateStoredValue (0 ms)
[----------] 1 test from StoredValueInstantiation_test/0 (0 ms total)

[----------] 1 test from StoredValueInstantiation_test/1, where TypeParam = float
[ RUN      ] StoredValueInstantiation_test/1.instantiateStoredValue
[       OK ] StoredValueInstantiation_test/1.instantiateStoredValue (0 ms)
[----------] 1 test from StoredValueInstantiation_test/1 (0 ms total)

[----------] 1 test from StoredValueInstantiation_test/2, where TypeParam = double
[ RUN      ] StoredValueInstantiation_test/2.instantiateStoredValue
[       OK ] StoredValueInstantiation_test/2.instantiateStoredValue (0 ms)
[----------] 1 test from StoredValueInstantiation_test/2 (0 ms total)

[----------] 1 test from StoredValueInstantiation_test/3, where TypeParam = signed char
[ RUN      ] StoredValueInstantiation_test/3.instantiateStoredValue
[       OK ] StoredValueInstantiation_test/3.instantiateStoredValue (0 ms)
[----------] 1 test from StoredValueInstantiation_test/3 (0 ms total)

[----------] 1 test from StoredValueInstantiation_test/4, where TypeParam = unsigned char
[ RUN      ] StoredValueInstantiation_test/4.instantiateStoredValue
[       OK ] StoredValueInstantiation_test/4.instantiateStoredValue (0 ms)
[----------] 1 test from StoredValueInstantiation_test/4 (0 ms total)

[----------] 1 test from StoredValueInstantiation_test/5, where TypeParam = short
[ RUN      ] StoredValueInstantiation_test/5.instantiateStoredValue
[       OK ] StoredValueInstantiation_test/5.instantiateStoredValue (0 ms)
[----------] 1 test from StoredValueInstantiation_test/5 (0 ms total)

[----------] 1 test from StoredValueInstantiation_test/6, where TypeParam = unsigned short
[ RUN      ] StoredValueInstantiation_test/6.instantiateStoredValue
[       OK ] StoredValueInstantiation_test/6.instantiateStoredValue (0 ms)
[----------] 1 test from StoredValueInstantiation_test/6 (0 ms total)

[----------] 1 test from StoredValueInstantiation_test/7, where TypeParam = int
[ RUN      ] StoredValueInstantiation_test/7.instantiateStoredValue
[       OK ] StoredValueInstantiation_test/7.instantiateStoredValue (0 ms)
[----------] 1 test from StoredValueInstantiation_test/7 (0 ms total)

[----------] 1 test from StoredValueInstantiation_test/8, where TypeParam = unsigned int
[ RUN      ] StoredValueInstantiation_test/8.instantiateStoredValue
[       OK ] StoredValueInstantiation_test/8.instantiateStoredValue (0 ms)
[----------] 1 test from StoredValueInstantiation_test/8 (0 ms total)

[----------] 1 test from StoredValueInstantiation_test/9, where TypeParam = __int64
[ RUN      ] StoredValueInstantiation_test/9.instantiateStoredValue
[       OK ] StoredValueInstantiation_test/9.instantiateStoredValue (0 ms)
[----------] 1 test from StoredValueInstantiation_test/9 (0 ms total)

[----------] 1 test from StoredValueInstantiation_test/10, where TypeParam = unsigned __int64
[ RUN      ] StoredValueInstantiation_test/10.instantiateStoredValue
[       OK ] StoredValueInstantiation_test/10.instantiateStoredValue (0 ms)
[----------] 1 test from StoredValueInstantiation_test/10 (0 ms total)

[----------] 1 test from StoredValueInstantiation_test/11, where TypeParam = class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >
[ RUN      ] StoredValueInstantiation_test/11.instantiateStoredValue
[       OK ] StoredValueInstantiation_test/11.instantiateStoredValue (0 ms)
[----------] 1 test from StoredValueInstantiation_test/11 (0 ms total)

[----------] 1 test from StoredValueInstantiation_test/12, where TypeParam = class std::array<int,4>
[ RUN      ] StoredValueInstantiation_test/12.instantiateStoredValue
[       OK ] StoredValueInstantiation_test/12.instantiateStoredValue (0 ms)
[----------] 1 test from StoredValueInstantiation_test/12 (0 ms total)

[----------] 10 tests from StoredValueTest
[ RUN      ] StoredValueTest.boolSet
[       OK ] StoredValueTest.boolSet (0 ms)
[ RUN      ] StoredValueTest.stringSet
[       OK ] StoredValueTest.stringSet (0 ms)
[ RUN      ] StoredValueTest.arrayGet
[       OK ] StoredValueTest.arrayGet (0 ms)
[ RUN      ] StoredValueTest.arraySet
[       OK ] StoredValueTest.arraySet (1 ms)
[ RUN      ] StoredValueTest.typesBool
D:\projects\cppexpose\source\tests\cppexpose-test\StoredValueTest.cpp(60): error: Value of: test
  Actual: true
Expected: false
[  FAILED  ] StoredValueTest.typesBool (0 ms)
[ RUN      ] StoredValueTest.typesSignedIntegral
D:\projects\cppexpose\source\tests\cppexpose-test\StoredValueTest.cpp(60): error: Value of: test
  Actual: true
Expected: false
[  FAILED  ] StoredValueTest.typesSignedIntegral (0 ms)
[ RUN      ] StoredValueTest.typesUnsignedIntegral
D:\projects\cppexpose\source\tests\cppexpose-test\StoredValueTest.cpp(60): error: Value of: test
  Actual: true
Expected: false
[  FAILED  ] StoredValueTest.typesUnsignedIntegral (1 ms)
[ RUN      ] StoredValueTest.typesString
D:\projects\cppexpose\source\tests\cppexpose-test\StoredValueTest.cpp(60): error: Value of: test
  Actual: true
Expected: false
[  FAILED  ] StoredValueTest.typesString (0 ms)
[ RUN      ] StoredValueTest.typesFloat
D:\projects\cppexpose\source\tests\cppexpose-test\StoredValueTest.cpp(60): error: Value of: test
  Actual: true
Expected: false
[  FAILED  ] StoredValueTest.typesFloat (1 ms)
[ RUN      ] StoredValueTest.typesArray
[       OK ] StoredValueTest.typesArray (0 ms)
[----------] 10 tests from StoredValueTest (3 ms total)

[----------] Global test environment tear-down
[==========] 128 tests from 42 test cases ran. (56 ms total)
[  PASSED  ] 118 tests.
[  FAILED  ] 10 tests, listed below:
[  FAILED  ] DirectValueTest.typesBool
[  FAILED  ] DirectValueTest.typesSignedIntegral
[  FAILED  ] DirectValueTest.typesUnsignedIntegral
[  FAILED  ] DirectValueTest.typesString
[  FAILED  ] DirectValueTest.typesFloat
[  FAILED  ] StoredValueTest.typesBool
[  FAILED  ] StoredValueTest.typesSignedIntegral
[  FAILED  ] StoredValueTest.typesUnsignedIntegral
[  FAILED  ] StoredValueTest.typesString
[  FAILED  ] StoredValueTest.typesFloat

10 FAILED TESTS
```

Visual Studio 2015 Update 3

**Comment by scheibel** [comment]
Do these tests still fail?

**Comment by j-o** [comment]
Yes.

**Comment by j-o** [comment]
The problem is the implementation of the test case which compares pointers to virtual member functions for equality. According to the standard (N4606 §5.10/3.3), the result of this is unspecified.

**Comment by scheibel** [comment]
My suggestion: re-implement the test case to not rely on method pointer comparison anymore.

**Comment by scheibel** [comment]
Closed with #24.
